pkg/miscellaneous/miscellaneous.go

```diff
@@ -28,16 +30,25 @@ func GetLatestFullSnapshotAndDeltaSnapList(store snapstore.SnapStore) (*snapstor
 		return nil, nil, err
 	}
 
+	metrics.SnapstoreLatestDeltasTotal.With(prometheus.Labels{}).Set(0)
+	metrics.SnapstoreLatestDeltasRevisionsTotal.With(prometheus.Labels{}).Set(0)
+
 	for index := len(snapList); index > 0; index-- {
 		if snapList[index-1].IsChunk {
 			continue
 		}
 		if snapList[index-1].Kind == snapstore.SnapshotKindFull {
 			sort.Sort(deltaSnapList)
+			metrics.SnapstoreLatestDeltasTotal.With(prometheus.Labels{}).Set(float64(len(deltaSnapList)))
```

Comment:
Both metrics should be updated at the end. Why are you updating insid this for loop. This will add incorrect points to both series.